internal/v1/handler_blueprints_test.go

```diff
@@ -196,6 +197,179 @@ func TestHandlers_ComposeBlueprint(t *testing.T) {
+	require.Equal(t, http.StatusCreated, respStatusCode)
+
+	var result []ComposeResponse
+	err = json.Unmarshal([]byte(body), &result)
+	require.NoError(t, err)
+	require.Len(t, result, 2)
+	require.Equal(t, ids[0], result[0].Id)
+	require.Equal(t, ids[1], result[1].Id)
+}
+
+func TestHandlers_ComposeBlueprint_with_One_Target(t *testing.T) {
+	ctx := context.Background()
+	ids := []uuid.UUID{uuid.New(), uuid.New()}
+	idx := 0
+	apiSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if "Bearer" == r.Header.Get("Authorization") {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+		require.Equal(t, "Bearer accesstoken", r.Header.Get("Authorization"))
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		result := composer.ComposeId{
+			Id: ids[idx],
+		}
+		idx += 1
+		encodeErr := json.NewEncoder(w).Encode(result)
+		require.NoError(t, encodeErr)
+	}))
+	defer apiSrv.Close()
+
+	dbase, err := dbc.NewDB()
+	require.NoError(t, err)
+	srv, tokenSrv := startServer(t, &testServerClientsConf{ComposerURL: apiSrv.URL}, &ServerConfig{
+		DBase:            dbase,
+		DistributionsDir: "../../distributions",
+	})
+	defer func() {
+		shutdownErr := srv.Shutdown(ctx)
+		require.NoError(t, shutdownErr)
+	}()
+	defer tokenSrv.Close()
+
+	id := uuid.New()
+	versionId := uuid.New()
+
+	uploadOptions := UploadRequest_Options{}
+	err = uploadOptions.FromAWSUploadRequestOptions(AWSUploadRequestOptions{
+		ShareWithAccounts: common.ToPtr([]string{"test-account"}),
+	})
+	require.NoError(t, err)
+	name := "Blueprint Human Name"
+	description := "desc"
+	blueprint := BlueprintBody{
+		Customizations: Customizations{
+			Packages: common.ToPtr([]string{"nginx"}),
+		},
+		Distribution: "centos-8",
+		ImageRequests: []ImageRequest{
+			{
+				Architecture: ImageRequestArchitectureX8664,
+				ImageType:    ImageTypesAws,
+				UploadRequest: UploadRequest{
+					Type:    UploadTypesAws,
+					Options: uploadOptions,
+				},
+			},
+			{
+				Architecture: ImageRequestArchitectureAarch64,
+				ImageType:    ImageTypesGuestImage,
+				UploadRequest: UploadRequest{
+					Type:    UploadTypesAwsS3,
+					Options: uploadOptions,
+				},
+			},
+		},
+	}
+	payload := ComposeBlueprintJSONBody{
+		ImageTypes: &[]ImageTypes{"aws"},
+	}
+	var message []byte
+	message, err = json.Marshal(blueprint)
+	require.NoError(t, err)
+	err = dbase.InsertBlueprint(ctx, id, versionId, "000000", "000000", name, description, message)
+	require.NoError(t, err)
+
+	respStatusCode, body := tutils.PostResponseBody(t, fmt.Sprintf("http://localhost:8086/api/image-builder/v1/experimental/blueprints/%s/compose", id.String()), payload)
+	require.Equal(t, http.StatusCreated, respStatusCode)
+
+	var result []ComposeResponse
+	err = json.Unmarshal([]byte(body), &result)
+	require.NoError(t, err)
+	require.Len(t, result, 1)
+	require.Equal(t, ids[0], result[0].Id)
+}
+
```

Comment:
I do not want to block you anymore, but would you mind taking a look later at [table driven testing](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests)? I think this test (and many others) might benefit from it. This needs a research, chances are it might be not suitable but I think this is worth exploring.

Reply:
I had to give it a try: https://github.com/osbuild/image-builder/pull/1206 :)
Let me know if you'd want to take over @mgold1234 ;)